src/main/java/randoop/generation/ConstantMining/TfIdfSelector.java

```diff
@@ -0,0 +1,131 @@
+package randoop.generation.ConstantMining;
+
+import java.util.HashMap;
+import java.util.Map;
+import randoop.sequence.Sequence;
+import randoop.util.Log;
+import randoop.util.Randomness;
+import randoop.util.SimpleList;
+
+/**
+ * This class selects a sequence based on TF-IDF. TfIdfSelector is only used when constant mining is
+ * enabled.
+ *
+ * <p>When the literal level is ClassOrInterfaceType or Package, TfIdfSelector is created and store
+ * the constant information inside its corresponding Class or Package, and when the literal level is
+ * ALL, TfIdfSelector stores all constants' information instead and only one global TfIdfSelector is
+ * created. By information, it means sequence frequency and number of occurrence.
+ */
+public class TfIdfSelector {
+
+  /** Map from a sequence to its corresponding weight based on TF-IDF. */
+  Map<Sequence, Double> constantWeight = new HashMap<>();
+
+  /** If true, output debugging information. */
+  private static final boolean DEBUG = false;
+
+  /**
+   * Initialize the TfIdfSelector with the frequency of the sequence, the number of classes that
+   * contain the sequence, and the total number of classes in the current scope.
+   *
+   * @param frequency map from sequence to its frequency
+   * @param classesWithConstant map from sequence to the number of classes in the current scope that
+   *     contain the sequence
```

Comment:
Document whether these maps are guaranteed to have the same keys, or they might have different keys.